core/frontend/src/RealityDataSource.ts

```diff
@@ -5,15 +5,48 @@
 /** @packageDocumentation
  * @module Tiles
  */
-import { request, RequestOptions } from "./request/Request";
-import { AccessToken, BentleyStatus, GuidString, Logger } from "@itwin/core-bentley";
-import { IModelError, OrbitGtBlobProps, RealityData, RealityDataFormat, RealityDataProvider, RealityDataSourceKey, RealityDataSourceProps } from "@itwin/core-common";
+import { GuidString, Logger } from "@itwin/core-bentley";
+import { Cartographic, EcefLocationProps, OrbitGtBlobProps, RealityData, RealityDataFormat, RealityDataProvider, RealityDataSourceKey } from "@itwin/core-common";
 import { FrontendLoggerCategory } from "./FrontendLoggerCategory";
-import { IModelApp } from "./IModelApp";
-import { CesiumIonAssetProvider, ContextShareProvider, getCesiumAccessTokenAndEndpointUrl, getCesiumAssetUrl, getCesiumOSMBuildingsUrl } from "./tile/internal";
+import { CesiumIonAssetProvider, ContextShareProvider, getCesiumAssetUrl } from "./tile/internal";
+import { RealityDataSourceTilesetUrlImpl } from "./RealityDataSourceTilesetUrlImpl";
+import { RealityDataSourceContextShareImpl } from "./RealityDataSourceContextShareImpl";
+import { RealityDataSourceCesiumIonAssetImpl } from "./RealityDataSourceCesiumIonAssetImpl";
+import { Range3dProps } from "@itwin/core-geometry";
 
-/**
- * This interface provide methods used to access a reality data from a reality data provider
+const loggerCategory: string = FrontendLoggerCategory.RealityData;
+
+/** This interface provide spatial location and volume of interest, in meters, centered around `spatial location`
+ * @alpha
+ */
+export interface SpatialLocationAndExtents {
+  /** location of the point at the center of the reaity data */
+  location: Cartographic | EcefLocationProps;
+  /** extents of the volume around location */
+  worldRange: Range3dProps;
+  /** true if this reality data is geolocated
+   * A reality data is geolocated when we can compute a valid position relative to the earth surface.
+   * Note that some reality data contains custom coordinate system without any information on the relative position on earth.
+   * These reality data will have a location and worldRange but will be identified by the isGeoLocated set to false.
+   */
+  isGeolocated: boolean;
+}
+
+/** This interface provides information to identify the product and engine that create this reality data
+ * @alpha
+ */
+export interface PublisherProductInfo {
+  /** product that create this reality data */
+  product: string;
+  /** engine that create this reality data */
+  engine: string;
+  /** the version of the engine that create this reality data */
+  version: string;
+  /** the number of children at the root of this reality data */
+  rootChildren?: string;
```

Comment:
This is "the number of children" but its type is `string`. What does this have to do with "publisher product info"?

Reply:
You are right. I've moved rootChildren from PublisherProductInfo to internal ThreeDTileFileInfo.
Change rootChildren type to number.
Add a new internal method on ThreeDTileFormatInterpreter to getFileInfo.